src/java/com/unifina/signalpath/blockchain/Web3jHelper.java

```diff
@@ -0,0 +1,249 @@
+		}
+
+		if (isArray) {
+			List arglist;
+			if (arg instanceof List) {
+				arglist = (List) arg;
+			} else if (arg.getClass().isArray()) {
+				arglist = arrayToList(arg);
+			} else {
+				throw new ClassCastException("Arg of type " + arg.getClass() + " should be a list to instantiate web3j Array");
+			}
+			Constructor listcons;
+			if (arraySize <= 0) {
+				listcons = DynamicArray.class.getConstructor(new Class[]{Class.class, List.class});
+			} else {
+				Class arrayClass = Class.forName("org.web3j.abi.datatypes.generated.StaticArray" + arraySize);
+				listcons = arrayClass.getConstructor(new Class[]{Class.class, List.class});
+			}
+			//create a list of transformed arguments
+			ArrayList transformedList = new ArrayList(arglist.size());
+			for (Object o : arglist) {
+				transformedList.add(instantiateType(type, o, false, -1));
+			}
+			return (Type) listcons.newInstance(type, transformedList);
+		}
+
+		Object constructorArg = null;
+		if (NumericType.class.isAssignableFrom(type)) {
+			constructorArg = asBigInteger(arg);
+		} else if (BytesType.class.isAssignableFrom(type)) {
+			if (arg instanceof byte[]) {
+				constructorArg = arg;
+			} else if (arg instanceof BigInteger) {
+				constructorArg = ((BigInteger) arg).toByteArray();
+			} else if (arg instanceof String) {
+				constructorArg = Numeric.hexStringToByteArray((String) arg);
+			}
+		} else if (Utf8String.class.isAssignableFrom(type)) {
+			constructorArg = arg.toString();
+		} else if (Address.class.isAssignableFrom(type)) {
+			constructorArg = arg.toString();
+		} else if (Bool.class.isAssignableFrom(type)) {
+			if (arg instanceof Boolean) {
+				constructorArg = arg;
+			} else {
+				BigInteger bival = asBigInteger(arg);
+				constructorArg = bival == null ? null : !bival.equals(BigInteger.ZERO);
+			}
+		}
+		if (constructorArg == null) {
+			throw new RuntimeException("Could not create type " + type + " from arg " + arg.toString() + " of type " + arg.getClass());
+		}
+		Constructor cons = type.getConstructor(new Class[]{constructorArg.getClass()});
+		return (Type) cons.newInstance(constructorArg);
+	}
+	/**
+	 *
+	 * @param web3j
+	 * @param erc20address address of ERC20 or 0x0 for ETH balance
+	 * @param holder_address
+	 * @return token balance in wei
+	 * @throws ExecutionException
+	 * @throws InterruptedException
+	 */
+	public static BigInteger getERC20Balance(Web3j web3j, String erc20address, String holder_address) throws ExecutionException, InterruptedException {
+		Address tokenAddress = new Address(erc20address);
+		if(tokenAddress.toUint160().getValue().equals(BigInteger.ZERO)){
+			EthGetBalance ethGetBalance = web3j
+					.ethGetBalance(holder_address, DefaultBlockParameterName.LATEST)
+					.sendAsync()
+					.get();
+			final BigInteger balance = ethGetBalance.getBalance();
+			return balance;
+		}
+		//    function balanceOf(address tokenOwner) public view returns (uint balance);
```

Comment:
remove dead code

Reply:
I think it's helpful to have this stub indicating the solidity method signature we're calling